formver.compiler-plugin/core/src/org/jetbrains/kotlin/formver/core/linearization/PureLinearizer.kt

```diff
@@ -46,17 +51,30 @@ class PureLinearizer(override val source: KtSourceElement?) : LinearizationConte
         throw PureLinearizerMisuseException("addStatement")
     }
 
-    override fun addDeclaration(decl: Declaration) {
-        throw PureLinearizerMisuseException("addDeclaration")
-    }
+    // Nothing to do here, as an assignment is also added
+    override fun addDeclaration(decl: Declaration) {}
 
     override fun addAssignment(lhs: ExpEmbedding, rhs: ExpEmbedding) {
-        throw PureLinearizerMisuseException("addAssignment")
+        // TODO: Move this to purity check
+        if (lhs.ignoringMetaNodes() !is VariableEmbedding) throw SnaktInternalException(
```

Comment:
The frequency with which you have to ignore meta nodes is a bit worrying.  Is there not a neater way of working with this?